### ISSUE TITLE
Make worms virus disable tools of bots?

### ISSUE THREAD

**Issue by rbcat** [colobot/colobot#1434]
Worms can mess with some variables in your programs if it infects your bot, but that all there is to it.

However, worms disable buildings entirely, so why can't we disable bots too? Allow them to move and fly, but disable their main tools.

Make it so that infected bots cannot use their main tool, whether it's manually or via program. Like, grabber cannot grab and drop, sniffer cannot sniff, shooter cannot shoot, shielder cannot shield, etc.

**Comment by Python-Sargent** [comment]
On some missions you _will_ have to go into worm territory, making it that you cannot use your robot may make you inevitably fail the mission, seeing that without your bot to carry/shoot things you cannot do particular parts of a mission. Ending in some problems of making it easy to not be able to accomplish the mission.

**Comment by ixPost** [comment]
Yes, on Virus(es) I thought they were dangerous, until I played a
free-game. ; |
